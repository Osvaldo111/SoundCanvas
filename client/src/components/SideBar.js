import React from "react";
import "../style/sideBar.css";
import BackgroundSideBar from "./sideBarsComponents/backgroundSideBar";
import SoundwaveColorBar from "./sideBarsComponents/soundwaveColorBar";
import SoundwaveThickBar from "./sideBarsComponents/soundwaveThickBar";
import SoundwaveWidthBar from "./sideBarsComponents/soundwaveWidthBar";
import TextBar from "./sideBarsComponents/textBar";
import closeButtn from "../images/closeButton.svg";
import { connect } from "react-redux";
import { displayMobileBttn } from "../actions";
import { isMobileBttnPress } from "../actions";
/**
 * author: Osvaldo Carrillo
 * Date: 02/08/2020
 * This class is designed to be the sidebar or the edition
 * bar in the application.
 */
class SideBar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      displaySideBar: "",
      displayBackgroundBar: false,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: false
    };
  }

  componentDidUpdate(prevProps) {
    //Check when the mobile bttn is press
    const currentMobileNavPress = this.props.canvasProps.checkBttnPress;
    const prevMobileNavPress = prevProps.canvasProps.checkBttnPress;

    if (currentMobileNavPress !== prevMobileNavPress) {
      if (currentMobileNavPress) {
        this.displayMobileSideBar();
        // Reducer init
        this.props.isMobileBttnPress(false);
      }
    }
  }

  closeMobileBar = () => {
    this.setState({
      displaySideBar: false,
      displayBackgroundBar: false,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: false,
      displaySecondSideBar: false
    });

    // Reducer to display the menu bttn in mobile
    this.props.displayMobileBttn(true);
  };

  closeSecondaryBarDesk = () => {
    this.setState({
      displaySideBar: false,
      displayBackgroundBar: false,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: false,
      displaySecondSideBar: false
    });
  };
  displayMobileSideBar = () => {
    this.setState({
      displaySideBar: true,
      // Init background bar
      displayBackgroundBar: true,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: false
    });
  };

  displayBackgroundBar = () => {
    this.setState({
      displaySecondSideBar: true,
      displayBackgroundBar: true,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: false
    });
  };

  displaySWColorBar = () => {
    this.setState({
      displaySecondSideBar: true,
      displayBackgroundBar: false,
      displaySWColorBar: true,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: false
    });
  };

  displaySWCThickBar = () => {
    this.setState({
      displaySecondSideBar: true,
      displayBackgroundBar: false,
      displaySWColorBar: false,
      displaySWThickBar: true,
      displaySWWidthBar: false,
      displayTextBar: false
    });
  };

  displaySWWidth = () => {
    this.setState({
      displaySecondSideBar: true,
      displayBackgroundBar: false,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: true,
      displayTextBar: false
    });
  };

  displayTextBar = () => {
    this.setState({
      displaySecondSideBar: true,
      displayBackgroundBar: false,
      displaySWColorBar: false,
      displaySWThickBar: false,
      displaySWWidthBar: false,
      displayTextBar: true
    });
  };

  render() {
    const { displaySideBar, displaySecondSideBar } = this.state;

    return (
      <div
        className="sideBarContainer"
        style={{ display: displaySideBar ? "flex" : "" }}
      >
        <div className="sideBarFirst">
          <p
            onClick={this.displayBackgroundBar}
            className="sideBarSelection"
            style={{
              backgroundColor: this.state.displayBackgroundBar ? "#434e5e" : ""
            }}
          >
            Background
          </p>
          <p onClick={this.displaySWColorBar} className="sideBarSelection">
            Sound Wave<br></br>Color
          </p>
          <p onClick={this.displaySWCThickBar} className="sideBarSelection">
            Sound Wave<br></br>Thickness
          </p>
          <p onClick={this.displaySWWidth} className="sideBarSelection">
            Sound Wave<br></br>Width
          </p>
          <p onClick={this.displayTextBar} className="sideBarSelection">
            Text
          </p>
        </div>
        <div
          className="sideBarSecond"
          style={{ display: displaySecondSideBar ? "flex" : "" }}
        >
          <BackgroundSideBar display={this.state.displayBackgroundBar} />
          <SoundwaveColorBar display={this.state.displaySWColorBar} />
          <SoundwaveThickBar display={this.state.displaySWThickBar} />
          <SoundwaveWidthBar display={this.state.displaySWWidthBar} />
          <TextBar display={this.state.displayTextBar} />
          <div className="sideBarCloseDesk">
            <img
              onClick={this.closeSecondaryBarDesk}
              src={closeButtn}
              alt=""
              className="sideBarClosebttn"
            />
          </div>
        </div>
        <div className="closeSideBar" onClick={this.closeMobileBar}></div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { canvasProps: state.canvas };
}
const mapDispatchToProps = {
  displayMobileBttn,
  isMobileBttnPress
};
export default connect(mapStateToProps, mapDispatchToProps)(SideBar);
