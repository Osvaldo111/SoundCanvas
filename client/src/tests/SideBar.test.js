import React from "react";
import ReactDOM from "react-dom";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SideBar from "../components/SideBar";
import { mount, shallow } from "enzyme";
//Redux
import { Provider } from "react-redux";
import store from "../reducers/store";
import { createSerializer } from "enzyme-to-json";
import { spy } from "sinon";
configure({ adapter: new Adapter() });

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Side bar <SideBar />", () => {
  it("renders correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <SideBar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
