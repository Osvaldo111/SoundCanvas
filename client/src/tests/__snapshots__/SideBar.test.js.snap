// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side bar <SideBar /> renders correctly 1`] = `
<Connect(SideBar)>
  <div
    className="sideBarContainer"
    style={
      Object {
        "display": "",
      }
    }
  >
    <div
      className="sideBarFirst"
    >
      <p
        className="sideBarSelection"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "",
          }
        }
      >
        Background
      </p>
      <p
        className="sideBarSelection"
        onClick={[Function]}
      >
        Sound Wave
        <br />
        Color
      </p>
      <p
        className="sideBarSelection"
        onClick={[Function]}
      >
        Sound Wave
        <br />
        Thickness
      </p>
      <p
        className="sideBarSelection"
        onClick={[Function]}
      >
        Sound Wave
        <br />
        Width
      </p>
      <p
        className="sideBarSelection"
        onClick={[Function]}
      >
        Text
      </p>
    </div>
    <div
      className="sideBarSecond"
      style={
        Object {
          "display": "",
        }
      }
    >
      <Connect(BackgroundSideBar)
        display={false}
      >
        <div
          className="backgroundBar"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="backSBTitle"
          >
            Colors
          </div>
          <div>
            <Connect(ColorPicker)
              handleGetColor={[Function]}
            >
              <div
                className="colorPickerContainer"
              >
                <ul
                  className="colorPickerlist"
                  onClick={[Function]}
                >
                  <li
                    data-color="#F56651"
                    style={
                      Object {
                        "backgroundColor": "#F56651",
                      }
                    }
                  />
                  <li
                    data-color="#7AE9C9"
                    style={
                      Object {
                        "backgroundColor": "#7AE9C9",
                      }
                    }
                  />
                  <li
                    data-color="#606A92"
                    style={
                      Object {
                        "backgroundColor": "#606A92",
                      }
                    }
                  />
                  <li
                    data-color="#FDCD47"
                    style={
                      Object {
                        "backgroundColor": "#FDCD47",
                      }
                    }
                  />
                  <li
                    data-color="#464849"
                    style={
                      Object {
                        "backgroundColor": "#464849",
                      }
                    }
                  />
                  <li
                    data-color="#D7D1D5"
                    style={
                      Object {
                        "backgroundColor": "#D7D1D5",
                      }
                    }
                  />
                  <li
                    data-color="#fff"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                      }
                    }
                  />
                </ul>
                <div
                  className="colorPickInput"
                >
                  <div>
                    Enter Hex Color
                  </div>
                  <div
                    className="contColPkHex"
                  >
                    <div
                      className="colorPickHex"
                      onClick={[Function]}
                    >
                      <p>
                        #
                      </p>
                    </div>
                    <input
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    style={
                      Object {
                        "color": "red",
                        "display": "none",
                      }
                    }
                  >
                    Please enter a color
                  </div>
                </div>
              </div>
            </Connect(ColorPicker)>
          </div>
        </div>
      </Connect(BackgroundSideBar)>
      <Connect(SoundwaveColorBar)
        display={false}
      >
        <div
          className="soundwaveColBar"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="swbarTitle"
          >
            Soundwave Colors
          </div>
          <div>
            <Connect(ColorPicker)
              handleGetColor={[Function]}
            >
              <div
                className="colorPickerContainer"
              >
                <ul
                  className="colorPickerlist"
                  onClick={[Function]}
                >
                  <li
                    data-color="#F56651"
                    style={
                      Object {
                        "backgroundColor": "#F56651",
                      }
                    }
                  />
                  <li
                    data-color="#7AE9C9"
                    style={
                      Object {
                        "backgroundColor": "#7AE9C9",
                      }
                    }
                  />
                  <li
                    data-color="#606A92"
                    style={
                      Object {
                        "backgroundColor": "#606A92",
                      }
                    }
                  />
                  <li
                    data-color="#FDCD47"
                    style={
                      Object {
                        "backgroundColor": "#FDCD47",
                      }
                    }
                  />
                  <li
                    data-color="#464849"
                    style={
                      Object {
                        "backgroundColor": "#464849",
                      }
                    }
                  />
                  <li
                    data-color="#D7D1D5"
                    style={
                      Object {
                        "backgroundColor": "#D7D1D5",
                      }
                    }
                  />
                  <li
                    data-color="#fff"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                      }
                    }
                  />
                </ul>
                <div
                  className="colorPickInput"
                >
                  <div>
                    Enter Hex Color
                  </div>
                  <div
                    className="contColPkHex"
                  >
                    <div
                      className="colorPickHex"
                      onClick={[Function]}
                    >
                      <p>
                        #
                      </p>
                    </div>
                    <input
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    style={
                      Object {
                        "color": "red",
                        "display": "none",
                      }
                    }
                  >
                    Please enter a color
                  </div>
                </div>
              </div>
            </Connect(ColorPicker)>
          </div>
        </div>
      </Connect(SoundwaveColorBar)>
      <Connect(SoundwaveThickBar)
        display={false}
      >
        <div
          className="soundwaveThickBar"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="swThickBarTitle"
          >
            Soundwave Thickness
          </div>
          <div
            className="swThickBarSliCon"
          >
            <Connect(SliderBttn)
              handleGetSliderValue={[Function]}
            >
              <div>
                <div
                  className="sliderBttnCtn"
                >
                  <input
                    id="myRange"
                    max="10"
                    min="1"
                    onChange={[Function]}
                    type="range"
                    value={1}
                  />
                </div>
                <p>
                  Value 
                  <span>
                    1
                  </span>
                </p>
              </div>
            </Connect(SliderBttn)>
          </div>
        </div>
      </Connect(SoundwaveThickBar)>
      <Connect(SoundwaveWidthBar)
        display={false}
      >
        <div
          className="soundwaveWidthBar"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div>
            <p
              className="swWidthkBarTitle"
            >
              Soundwave Width
            </p>
          </div>
          <div
            className="swWidthBarSliCon"
          >
            <Connect(SliderBttn)
              handleGetSliderValue={[Function]}
            >
              <div>
                <div
                  className="sliderBttnCtn"
                >
                  <input
                    id="myRange"
                    max="10"
                    min="1"
                    onChange={[Function]}
                    type="range"
                    value={1}
                  />
                </div>
                <p>
                  Value 
                  <span>
                    1
                  </span>
                </p>
              </div>
            </Connect(SliderBttn)>
          </div>
        </div>
      </Connect(SoundwaveWidthBar)>
      <Connect(TextBar)
        display={false}
      >
        <div
          className="textBar"
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="textBarTitle"
          >
            Text
          </div>
          <div>
            <div>
              Click text to add to page
            </div>
            <div
              className="txtBarBdyTxt"
              onClick={[Function]}
            >
              <p>
                Add some body text
              </p>
            </div>
          </div>
          <div
            className="txtBarColFnt"
          >
            <div>
              Select font color
            </div>
            <Connect(ColorPicker)
              handleGetColor={[Function]}
            >
              <div
                className="colorPickerContainer"
              >
                <ul
                  className="colorPickerlist"
                  onClick={[Function]}
                >
                  <li
                    data-color="#F56651"
                    style={
                      Object {
                        "backgroundColor": "#F56651",
                      }
                    }
                  />
                  <li
                    data-color="#7AE9C9"
                    style={
                      Object {
                        "backgroundColor": "#7AE9C9",
                      }
                    }
                  />
                  <li
                    data-color="#606A92"
                    style={
                      Object {
                        "backgroundColor": "#606A92",
                      }
                    }
                  />
                  <li
                    data-color="#FDCD47"
                    style={
                      Object {
                        "backgroundColor": "#FDCD47",
                      }
                    }
                  />
                  <li
                    data-color="#464849"
                    style={
                      Object {
                        "backgroundColor": "#464849",
                      }
                    }
                  />
                  <li
                    data-color="#D7D1D5"
                    style={
                      Object {
                        "backgroundColor": "#D7D1D5",
                      }
                    }
                  />
                  <li
                    data-color="#fff"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                      }
                    }
                  />
                </ul>
                <div
                  className="colorPickInput"
                >
                  <div>
                    Enter Hex Color
                  </div>
                  <div
                    className="contColPkHex"
                  >
                    <div
                      className="colorPickHex"
                      onClick={[Function]}
                    >
                      <p>
                        #
                      </p>
                    </div>
                    <input
                      onChange={[Function]}
                      required={true}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    style={
                      Object {
                        "color": "red",
                        "display": "none",
                      }
                    }
                  >
                    Please enter a color
                  </div>
                </div>
              </div>
            </Connect(ColorPicker)>
          </div>
          <div>
            <Connect(FontSizeSelector)
              handleGetFontSize={[Function]}
            >
              <div>
                <div>
                  Font Size
                </div>
                <div
                  className="ftSelConIn"
                  onClick={[Function]}
                >
                  <input
                    onChange={[Function]}
                    onKeyUp={[Function]}
                    type="number"
                    value="14"
                  />
                  <div
                    className="ftSelConInIm"
                  >
                    <img
                      alt=""
                      src="arrowDown.svg"
                    />
                  </div>
                </div>
                <div
                  className="ftSelCtnDrpDwn"
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                >
                  <ul
                    onClick={[Function]}
                  >
                    <li
                      data-size="6"
                    >
                      6
                    </li>
                    <li
                      data-size="8"
                    >
                      8
                    </li>
                    <li
                      data-size="10"
                    >
                      10
                    </li>
                    <li
                      data-size="12"
                    >
                      12
                    </li>
                    <li
                      data-size="14"
                    >
                      14
                    </li>
                    <li
                      data-size="16"
                    >
                      16
                    </li>
                    <li
                      data-size="18"
                    >
                      18
                    </li>
                    <li
                      data-size="21"
                    >
                      21
                    </li>
                    <li
                      data-size="24"
                    >
                      24
                    </li>
                    <li
                      data-size="28"
                    >
                      28
                    </li>
                    <li
                      data-size="32"
                    >
                      32
                    </li>
                    <li
                      data-size="36"
                    >
                      36
                    </li>
                    <li
                      data-size="42"
                    >
                      42
                    </li>
                    <li
                      data-size="48"
                    >
                      48
                    </li>
                    <li
                      data-size="56"
                    >
                      56
                    </li>
                  </ul>
                </div>
              </div>
            </Connect(FontSizeSelector)>
          </div>
        </div>
      </Connect(TextBar)>
      <div
        className="sideBarCloseDesk"
      >
        <img
          alt=""
          className="sideBarClosebttn"
          onClick={[Function]}
          src="closeButton.svg"
        />
      </div>
    </div>
    <div
      className="closeSideBar"
      onClick={[Function]}
    />
  </div>
</Connect(SideBar)>
`;
